pkg/api/validation/validation.go

```diff
@@ -731,6 +731,14 @@ func ValidateService(service *api.Service) errs.ValidationErrorList {
 		allErrs = append(allErrs, errs.NewFieldNotSupported("spec.sessionAffinity", service.Spec.SessionAffinity))
 	}
 
+	for _, ip := range service.Spec.PublicIPs {
+		if !util.IsValidIP(ip) || ip == "0.0.0.0" {
+			allErrs = append(allErrs, errs.NewFieldInvalid("spec.publicIPs", ip, "is not an IP address"))
+		} else if ip == "127.0.0.1" {
```

Comment:
How about 

```
if net.ParseIP(ip).IsLoopback() 
```

Being loopback is more than just 127.0.0.1.

In fact, http://golang.org/pkg/net/#ParseIP has a bunch of conditions that are probably worth checking.  Maybe

```
if !net.ParseIP(ip).IsGlobalUnicast() {
```